server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2375,6 +2376,7 @@ protected void commitIndexWriter(final IndexWriter writer, final Translog transl
                     commitData.put(FORCE_MERGE_UUID_KEY, currentForceMergeUUID);
                 }
                 commitData.put(Engine.MIN_RETAINED_SEQNO, Long.toString(softDeletesPolicy.getMinRetainedSeqNo()));
+                commitData.put(ES_VERSION, Version.CURRENT.toString());
```

Comment:
I did not see the two assertions added, perhaps there were issues doing so? If so, please let me know.

Reply:
Sorry I missed that comment.

> Can we perhaps add an assertion about the previous version being before or on current version?

Do you mean asserting that the commit user data that doesn't contain the `ES_VERSION` value is in a version < 7.16? @henningandersen

Reply:
My intention with that specific comment was that if there is a version, it must be onOrBefore current version. I.e., the version can only go forward never backwards. This will be verified a little bit once we do some rolling upgrade tests between 7.x and 8.0 (i.e., after merge to both). And obviously as we add releases this check will be more valuable. Just want it to be expressed in an assertion that it is monotonically increasing.